qt/scientific_interfaces/Indirect/IndirectFitPlotView.cpp

```diff
@@ -46,27 +46,27 @@ std::string IndirectFitPlotView::getSpectrumText() const {
   return m_plotForm->cbPlotSpectrum->currentText().toStdString();
 }
 
-std::size_t IndirectFitPlotView::getSelectedSpectrum() const {
+WorkspaceIndex IndirectFitPlotView::getSelectedSpectrum() const {
   if (m_plotForm->swPlotSpectrum->currentIndex() == 0)
-    return m_plotForm->spPlotSpectrum->value();
+    return WorkspaceIndex{m_plotForm->spPlotSpectrum->value()};
   else if (m_plotForm->cbPlotSpectrum->count() != 0)
-    return std::stoull(getSpectrumText());
-  return 0;
+    return WorkspaceIndex{std::stoi(getSpectrumText())};
```

Comment:
Do we verify in `getSpectrumText` that we always get a numeric value back?

Reply:
The string here comes from a combobox which will only ever have numbers in as we set it's values